pkg/location/type_git.go

```diff
@@ -127,15 +170,34 @@ func (ref Git) GetDefaultDirectoryName() (string, bool) {
 	return path.Base(path.Join(path.Clean(repo), path.Clean(ref.Directory))), true
 }
 
-// SetIdentifier is called from mutate.Identifier
-func (ref Git) SetIdentifier(name string) (Reference, error) {
+func (ref Git) DefaultRevision(ctx context.Context) (string, error) {
+	gur, err := gitutil.NewGitUpstreamRepo(ctx, ref.Repo)
+	if err != nil {
+		return "", err
+	}
+	b, err := gur.GetDefaultBranch(ctx)
```

Comment:
That could be a question for @droot --- it's true it's the only place that the reference/parsing layer and it's only because in the existing CLI will fully qualify the git repo (adding default branch name) if it's not provided.

That said --- **Open** on a reference does return _both_ the ref and the refLock for that reason. The CLI could be changed to allow `location.Git{Ref string}` at parse-time remain empty-string to imply the default branch --- and the Ref returned could be treated as the correct value to persist in Kptfile.upstream